machineconfiguration/v1/types.go

```diff
@@ -30,6 +30,10 @@ type ControllerConfig struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
+	// TODO(jkyros): inconsistent historical generation resulted in the controllerconfig CRD being
+	// generated with all fields required, while everything else was generated with optional
+
+	// +kubebuilder:validation:Required
 	// +required
```

Comment:
This tag isn't correct, please drop the `+required` from everywhere, only use `+kubebuilder:validation:Required`

Reply:
thanks, removed all `+required`